apps/hash-graph/bin/cli/src/args.rs

```diff
@@ -1,11 +1,16 @@
 use clap::Parser;
 
-use crate::subcommand::Subcommand;
+use crate::{parser::OptionalSentryDsnParser, subcommand::Subcommand};
 
 /// Arguments passed to the program.
 #[derive(Debug, Parser)]
 #[clap(version, author, about, long_about = None)]
 pub struct Args {
+    // we need to qualify `Option` here, as otherwise `clap` tries to be too smart and only uses
+    // the `value_parser` on the internal `sentry::types::Dsn`, failing.
+    #[arg(long, env = "HASH_GRAPH_SENTRY_DSN", value_parser = OptionalSentryDsnParser)]
+    pub sentry_dsn: core::option::Option<sentry::types::Dsn>,
+
```

Comment:
I wonder if this is supposed to be a global argument. However, the only service where it does not matter is the `completions` endpoint I'd assume. (non-blocking comment)

Reply:
Yes, you are right, but (as you noted), I have decided to move it to a global flag, as the only command that does not benefit is the completions command, and it would lead to more code duplication (and make the guard more complicated)

> In my mind it also makes sense when you `--help`, as something like sentry or tracing should likely be a global argument and come before any subcommand. That way it is easier to switch out commands, etc.

Reply:
That's fair. Do you know if it's possible to exclude this argument for certain subcommands? (Just wondering, doesn't block this PR)

> something like sentry or tracing should likely be a global argument and come before any subcommand

It's possible to achieve this with `clap` when making this a `global` argument.